src/couch_replicator_manager.erl

```diff
@@ -170,7 +173,13 @@ handle_call({owner, RepId}, _From, State) ->
         {reply, owner(DbName, DocId, State#state.live), State}
     end;
 
-handle_call({rep_db_update, DbName, {ChangeProps} = Change}, _From, State) ->
+% Match changes epoch with the current epoch in the state.
+% New epoch ref is created on a full rescan. Change feeds have to
+% be replayed from the start to determine ownership in the new
+% cluster configuration and epoch is used to match & checkpoint
+% only change feeds from the current cluster configuration.
+handle_call({rep_db_update, DbName, {ChangeProps} = Change, Epoch},
+            _From, #state{epoch = Epoch} = State) ->
```

Comment:
I'm not convinced we should guard on rep_db_update messages unless I'm forgetting that we mark that in the ets table. But as far as I recall we only record during the rep_db_checkpoint message. Not matching should theoretically only be an issue in that we process some updates multiple times which shouldn't be an issue. I'm can't see the obvious upside of preventing that and I'm not able to immediately guarantee there's not a downside. So I'd rather leave to the existing behavior rather than get too far ahead of ourselves.

Replicator bugs are scary. Best not to do much at once here.